src/tests/ftest/pool/create_capacity.py

```diff
@@ -20,12 +22,284 @@ class PoolCreateCapacityTests(TestWithServers):
+
+            for scm_device in host_storage["storage"]["scm_namespaces"]:
+                rank = scm_device["mount"]["rank"]
+                if rank not in scm_engine_bytes:
+                    scm_engine_bytes[rank] = 0
+                scm_engine_bytes[rank] += scm_device["mount"]["avail_bytes"]
+
+            if host_storage["storage"]["nvme_devices"] is None:
+                continue
+
+            for nvme_device in host_storage["storage"]["nvme_devices"]:
+                if nvme_device["smd_devices"] is None:
+                    continue
+                namespace_id = 0
+                for smd_device in nvme_device["smd_devices"]:
+                    namespace_id += 1
+                    if smd_device["dev_state"] != "NORMAL":
+                        continue
+                    rank = smd_device["rank"]
+                    roles = smd_device["roles"]
+                    if not roles:
+                        # md-on-scm has only one implicit data role
+                        roles = "data"
+                    if rank not in nvme_engine_bytes:
+                        nvme_engine_bytes[rank] = {}
+                    if roles not in nvme_engine_bytes[rank]:
+                        nvme_engine_bytes[rank][roles] = 0
+                    namespaces = nvme_device["namespaces"]
+                    cluster_bytes = smd_device["cluster_size"]
+                    nvme_engine_bytes[rank][roles] += self.get_nsp_available_bytes(
+                        namespaces, namespace_id, cluster_bytes)
+
+        scm_bytes = sys.maxsize
+        for size in scm_engine_bytes.values():
+            scm_bytes = min(scm_bytes, size)
+        if scm_bytes == sys.maxsize:
+            scm_bytes = 0
+
+        nvme_bytes = {}
+        for it in nvme_engine_bytes.values():
+            for roles, roles_bytes in it.items():
+                if roles not in nvme_bytes:
+                    nvme_bytes[roles] = roles_bytes
+                nvme_bytes[roles] = min(nvme_bytes[roles], roles_bytes)
+
+        return scm_bytes, nvme_bytes
+
+    def get_nvme_attr(self, storage_usage, attr_name):
+        """Returns the first json value of a given NVMe device attribute.
+
+        Args:
+            storage_usage (dict): dmg json command output defining the storage usage.
+            attr_name (str): Name of a NVMe devices attribute.
+
+        Returns:
+            object: json value of given attribute.
+
+        """
+        for host_storage in storage_usage["response"]["HostStorage"].values():
+            for nvme_device in host_storage["storage"]["nvme_devices"]:
+                if nvme_device["smd_devices"] is None:
+                    continue
+                for smd_device in nvme_device["smd_devices"]:
+                    if smd_device["dev_state"] != "NORMAL":
+                        continue
+                    return smd_device[attr_name]
+        raise AttributeError
```

Comment:
Minor - It would be better for this to have a description
```suggestion
        raise AttributeError("Reason why")
```

Reply:
- [ ] Add message in exception

Reply:
Fixed with commit 01eea32a89491faed5dbf7f03085398ff402c3c6